sympy/physics/control/lti.py

```diff
@@ -1144,3 +1325,195 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
 
     def __neg__(self):
         return Feedback(-self.num, self.den)
+
+
+class TransferFunctionMatrix(Basic):
+
+    def __new__(cls, arg, shape=None, var=None):
+        if not (isinstance(arg, (list, tuple)) and
+            (all(isinstance(i, (list, tuple)) for i in arg) or
+            all(isinstance(i, (TransferFunction, Parallel, Series)) for i in arg))):
+            raise TypeError("Unsupported type for argument of TransferFunctionMatrix.")
+
+        if shape and var:
+            if not isinstance(shape, tuple):
+                raise TypeError("Shape must be a tuple, not {}.".format(type(shape)))
+            if not isinstance(var, Symbol):
+                raise TypeError("Var must be a Symbol, not {}.".format(type(var)))
+
+        obj = super(TransferFunctionMatrix, cls).__new__(cls, arg)
+        if all(isinstance(i, (TransferFunction, Series, Parallel)) for i in arg):
+            # TFM with 1st argument of the type - [tf1, tf2, tf3, ...] or (tf1, tf2, ...)
+            if shape and var:
+                if not all(elem.var == var for elem in arg):
+                    raise ValueError("All transfer functions should use the same complex"
+                        " variable (var) of the Laplace transform.")
+                if not (shape[0] == len(arg) and shape[1] == 1):
+                    raise ValueError("Shape must be equal to ({0}, {1}).".format(len(arg), 1))
```

Comment:
A better message might be "The provided Shape does not match the shape of the input. Shape must be equal to ..."

Reply:
Same at other instances